src/python/pants/notes/master.rst

```diff
@@ -4,6 +4,152 @@ Master Pre-Releases
+
+Refactoring, Improvements, and Tooling
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+* Use f-strings in `src/python/pants/pantsd/pailgun_server.py` + fix typos in comments. (#8949)
+  `PR #8949 <https://github.com/pantsbuild/pants/pull/8949>`_
+
+* Add some type annotations to `goal/` and `auth/` (#8934)
+  `PR #8934 <https://github.com/pantsbuild/pants/pull/8934>`_
+
+* Use f-strings on `src/python/pants/testutil` and `src/python/pants/util/` (#8950)
+  `PR #8950 <https://github.com/pantsbuild/pants/pull/8950>`_
+
+* Rename public Target methods to use `address_spec` instead of `spec` (#8944)
+  `PR #8944 <https://github.com/pantsbuild/pants/pull/8944>`_
+
+* Only compile non-module exports for `export-dep-as-jar` goal (#8914)
+  `PR #8914 <https://github.com/pantsbuild/pants/pull/8914>`_
+
+* Address some remaining issues with setup-py2. (#8927)
+  `PR #8927 <https://github.com/pantsbuild/pants/pull/8927>`_
+
+* Tweak the argument parser to allow file system specs (#8932)
+  `PR #8932 <https://github.com/pantsbuild/pants/pull/8932>`_
+
+* Share the local LMDB store to avoid creating one per test method. (#8933)
+  `PR #8933 <https://github.com/pantsbuild/pants/pull/8933>`_
+
+* Allow `python_test_runner.py` to get coverage for individual targets (#8910)
+  `PR #8910 <https://github.com/pantsbuild/pants/pull/8910>`_
+
+* Fix confusing error for an `@console_rule` not returning a `Goal` (#8931)
+  `PR #8931 <https://github.com/pantsbuild/pants/pull/8931>`_
+
+* Have handle_workunits callback use kwargs (#8918)
+  `PR #8918 <https://github.com/pantsbuild/pants/pull/8918>`_
+
+* Deprecate `--positional-arg-file` in favor of `--spec-file` (#8928)
+  `PR #8928 <https://github.com/pantsbuild/pants/pull/8928>`_
+
+* Refactor `options_test.py` (#8906)
+  `PR #8906 <https://github.com/pantsbuild/pants/pull/8906>`_
+
+* Simplify `./pants test --debug` requesting one single address (#8925)
+  `PR #8925 <https://github.com/pantsbuild/pants/pull/8925>`_
+
+* Stop adding our own colors to V2 test output (#8926)
+  `PR #8926 <https://github.com/pantsbuild/pants/pull/8926>`_
+
+* Update core test model to include coverage results (#8851)
+  `PR #8851 <https://github.com/pantsbuild/pants/pull/8851>`_
+
+* Use color in v2 pytest (#8901)
+  `PR #8901 <https://github.com/pantsbuild/pants/pull/8901>`_
+
+* Hotfix Specs rename conflicting with setup-py2 change (#8920)
+  `PR #8920 <https://github.com/pantsbuild/pants/pull/8920>`_
+
+* Prepare 1.24.0rc1. (#8919)
+  `PR #8919 <https://github.com/pantsbuild/pants/pull/8919>`_
+
+* Rename `Spec` to `AddressSpec` (#8916)
+  `PR #8916 <https://github.com/pantsbuild/pants/pull/8916>`_
+
+* Bump up deprecations for the 1.26 series to earlier in the 1.26 dev series (#8908)
+  `PR #8908 <https://github.com/pantsbuild/pants/pull/8908>`_
+
+* Add type hints to `specs.py` (#8913)
+  `PR #8913 <https://github.com/pantsbuild/pants/pull/8913>`_
+
+* Rename `optionable_rule` to `subsystem_rule` (#8893)
```

Comment:
This should also be an API change, given that `console_rule -> goal_rule` is classified as that.